### PR TITLE
[Merged by Bors] - feat: port MeasureTheory.PiSystem

### PR DESCRIPTION
**Pull request by Komyyy**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#2105
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by Komyyy** [comment]
`a₁ ⊔ a₂ =?= max a₁ a₂` fails. Once wait for #2105.

**Comment by Komyyy** [comment]
I tried merging #2105, but `a₁ ⊔ a₂ =?= max a₁ a₂` failed again.

**Comment by Komyyy** [comment]
Once wait for leanprover/lean4#2003.

**Comment by urkud** [comment]
I fixed compile by using `sup`/`inf` in `Ixx` lemmas.

**Comment by Komyyy** [comment]
@urkud Thank you!
@joneugster I mistook a **mathlib-port** label for a **WIP** label. Thank you!

File: Mathlib/MeasureTheory/PiSystem.lean [hunk 1]
```diff
@@ -0,0 +1,767 @@
...
+  A Dynkin system is also known as a "λ-system" or a "d-system".
+-/
+structure DynkinSystem (α : Type _) where
+  /-- Predicate saying that a given set is contained in the dynkin system. -/
+  Has : Set α → Prop
+  /-- A dynkin system contains the empty set. -/
+  has_empty : Has ∅
+  /-- A dynkin system is closed under complementation. -/
+  has_compl : ∀ {a}, Has a → Has (aᶜ)
+  /-- A dynkin system is closed under countable union of pairwise disjoint sets. Use a more general
```
Comment (jcommelin) [inline Mathlib/MeasureTheory/PiSystem.lean:560]:
```suggestion
  /-- Predicate saying that a given set is contained in the Dynkin system. -/
  Has : Set α → Prop
  /-- A Dynkin system contains the empty set. -/
  has_empty : Has ∅
  /-- A Dynkin system is closed under complementation. -/
  has_compl : ∀ {a}, Has a → Has (aᶜ)
  /-- A Dynkin system is closed under countable union of pairwise disjoint sets. Use a more general
```

File: Mathlib/MeasureTheory/PiSystem.lean [hunk 2]
```diff
@@ -0,0 +1,767 @@
...
+
+theorem isPiSystem_image_Ioi (s : Set α) : IsPiSystem (Ioi '' s) :=
+  @isPiSystem_image_Iio αᵒᵈ _ s
+#align is_pi_system_image_Ioi isPiSystem_image_Ioi
+
+theorem isPiSystem_Ioi : IsPiSystem (range Ioi : Set (Set α)) :=
+  @image_univ α _ Ioi ▸ isPiSystem_image_Ioi univ
+#align is_pi_system_Ioi isPiSystem_Ioi
+
+-- porting note: changed `max`/`min` to `⊔`/`⊓`
```
Comment (jcommelin) [inline Mathlib/MeasureTheory/PiSystem.lean:155]: Was there a Zulip discussion about this? If not, I started one here: https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/MeasureTheory.2EPiSystem.20!4.232160

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Pull request closed without merging**